addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/accounts/AwsAccountsList.js

```diff
@@ -38,16 +38,55 @@ class AwsAccountsList extends React.Component {
       // and value as flag indicating whether to show the editor for the user
       this.mapOfUsersBeingEdited = {};
       this.selectedFilter = 'All'; // case-sensitive, see AwsAccountsStore.js for options
+      this.awsAccountIdsOfActiveEnvs = [];
     });
   }
 
-  componentDidMount() {
+  async componentDidMount() {
     const accountsStore = this.props.accountsStore;
     const awsAccountsStore = this.props.awsAccountsStore;
     swallowError(accountsStore.load());
     swallowError(awsAccountsStore.load());
     accountsStore.startHeartbeat();
     awsAccountsStore.startHeartbeat();
+    const awsAccountIdOfActiveEnvs = await this.getAccountsWithActiveEnvironments();
+    runInAction(() => {
+      this.awsAccountIdsOfActiveEnvs = awsAccountIdOfActiveEnvs;
+    });
+  }
+
+  /**
+   * @return List of account ids of all accounts with active environments
+   */
+  async getAccountsWithActiveEnvironments() {
+    const scEnvironmentStore = this.props.scEnvironmentsStore;
```

Comment:
Could we add getter methods for each of these stores, and decorate them as `computed`? This would allow us to clean up the `Promise.all()` type initialization below.

Reply:
Note, I'm moving this logic to check for active account every time a user clicks the `Re-Onboard Account` button. This is to prevent the scenario where a new workspace is spun up before the user decides to click the `Re-Onboard Account` button.

On that note, if we used `computed`, will `mobx` cache the API results? We want live data and not any cached results.

Info [here](https://mobx.js.org/computeds.html)
`Computed values can be used to derive information from other observables. They evaluate lazily, caching their output and only recomputing if one of the underlying observables has changed.`